packages/tailwindcss/src/theme.ts

```diff
@@ -158,8 +183,9 @@ export class Theme {
     candidateValue: string,
     themeKeys: ThemeKey[],
     nestedKeys: `--${string}`[] = [],
+    ignoredThemeKeys: ThemeKey[] = [],
   ): [string, Record<string, string>] | null {
-    let themeKey = this.#resolveKey(candidateValue, themeKeys)
+    let themeKey = this.#resolveKey(candidateValue, themeKeys, ignoredThemeKeys)
```

Comment:
We don't have any automated tests that cover this because there is nowhere in Tailwind that needs to use `ignoredThemeKeys` that _also_ depends on nested theme keys, so to make sure this works and is necessary Jordan and I manually tested this by updating the `text` utility to ignore the `--font-size-potato` key and added `--font-size-potato-huge` to the theme.

Replacing `ignoredThemeKeys` here with an empty array caused a `text-potato-huge` class to be generated mistakenly, and putting `ignoredThemeKeys` back caused it to not be generated as expected.